creusot/tests/should_succeed/result/own.rs

```diff
@@ -0,0 +1,213 @@
+        match self {
+            #[allow(unused_variables)]
+            OwnResult::Ok(x) => None,
+            OwnResult::Err(x) => Some(x),
+        }
+    }
+
+    #[ensures(forall<t: &T> *self == OwnResult::Ok(*t) ==> result == OwnResult::Ok(t))]
+    #[ensures(forall<e: &E> *self == OwnResult::Err(*e) ==> result == OwnResult::Err(e))]
+    #[ensures((exists<t: &T> *self == OwnResult::Ok(*t))|| (exists<e: &E> *self == OwnResult::Err(*e)))]
+    pub fn as_ref(&self) -> OwnResult<&T, &E> {
+        match *self {
+            OwnResult::Ok(ref x) => OwnResult::Ok(x),
+            OwnResult::Err(ref x) => OwnResult::Err(x),
+        }
+    }
+
+    #[ensures(
+        forall<t: &mut T> *self == OwnResult::Ok(*t) ==> ^self == OwnResult::Ok(^t) && result == OwnResult::Ok(t)
+    )]
+    #[ensures(
+        forall<e: &mut E> *self == OwnResult::Err(*e) ==> ^self == OwnResult::Err(^e) && result == OwnResult::Err(e)
+    )]
+    #[ensures(
+        (exists<t: &mut T> *self == OwnResult::Ok(*t))|| (exists<e: &mut E> *self == OwnResult::Err(*e))
+    )]
+    pub fn as_mut(&mut self) -> OwnResult<&mut T, &mut E> {
+        match *self {
+            OwnResult::Ok(ref mut x) => OwnResult::Ok(x),
+            OwnResult::Err(ref mut x) => OwnResult::Err(x),
+        }
+    }
+
+    #[requires(exists<t: T> self == OwnResult::Ok(t))]
+    #[ensures(OwnResult::Ok(result) == self)]
+    pub fn unwrap(self) -> T
+    where
+        E: ::std::fmt::Debug,
+    {
+        match self {
+            OwnResult::Ok(t) => t,
+            OwnResult::Err(_e) => panic!(),
+        }
+    }
+
+    #[requires(exists<t: T> self == OwnResult::Ok(t))]
+    #[ensures(OwnResult::Ok(result) == self)]
+    pub fn expect(self, msg: &str) -> T
+    where
+        E: ::std::fmt::Debug,
+    {
+        match self {
+            OwnResult::Ok(t) => t,
+            OwnResult::Err(_e) => panic!(),
+        }
+    }
+
+    #[requires(exists<e: E> self == OwnResult::Err(e))]
+    #[ensures(OwnResult::Err(result) == self)]
+    pub fn unwrap_err(self) -> E
+    where
+        T: ::std::fmt::Debug,
+    {
+        match self {
+            OwnResult::Ok(_t) => panic!(),
+            OwnResult::Err(e) => e,
+        }
+    }
+
+    #[ensures(forall<t: T> self == OwnResult::Ok(t) ==> result == t)]
+    #[ensures((exists<e: E> self == OwnResult::Err(e)) ==> result == default)]
+    pub fn unwrap_or(self, default: T) -> T {
+        match self {
+            OwnResult::Ok(t) => t,
+            #[allow(unused_variables)]
+            OwnResult::Err(e) => default,
+        }
+    }
+
+    //#[ensures(forall<t: T> self == OwnResult::Ok(t) ==> result == t)]
+    //#[ensures((exists<e: E> self == OwnResult::Err(e)) ==> result.is_default())]
+    //pub fn unwrap_or_default(self) -> T
+    //where
+    //    T: Default,
+    //{
+    //    match self {
+    //        OwnResult::Ok(x) => x,
+    //        OwnResult::Err(_) => Default::default(),
+    //    }
+    //}
```

Comment:
Not sure how to make this work.

Reply:
It seems like here the error for whatever reason is the `Default::default`. Replacing this by `T::default()` fixes it.